Add Crown Jewels docs

jakule opened the pull request:
Add Crown Jewels documentation and bump TAG version in docs to the latest.

Closes https://github.com/gravitational/access-graph/issues/1194

jakule committed:
Add Crown Jewels docs

jakule committed:
Update role definition

zmb3 commented on docs/img/access-graph/crown-jewels/create-matcher-name.png:
I recognize this is only a docs PR, but wanted to note that Teleport almost always puts the primary action button on the left, and the close/cancel button on the right.

Being consistent is important for muscle memory. Nothing is more frustrating than mistakenly clicking cancel because it keeps changing places on you.

<img width="345" alt="Screenshot 2024-09-19 at 2 00 07 PM" src="https://github.com/user-attachments/assets/5abae05c-f60e-470b-8b2a-b1083b977508">


<img width="373" alt="Screenshot 2024-09-19 at 2 00 18 PM" src="https://github.com/user-attachments/assets/219120ba-9678-4071-91cd-22228f463110">

<img width="343" alt="Screenshot 2024-09-19 at 2 00 31 PM" src="https://github.com/user-attachments/assets/a5848369-efc5-4207-a34f-f4bf456d67c3">

jakule replied:
CC @ryanclark

ryanclark replied:
<img width="471" alt="image" src="https://github.com/user-attachments/assets/f0009970-9b1d-4d41-9044-4260bcf50304">

Removed the cancel button as there's an `X` to close the modal. I don't really agree with the cancel button being on the right (I get that Teleport does it, and if we have both, we should be consistent), and left-aligning the buttons just to put the cancel button on the right looks strange.

zmb3 commented on docs/img/access-graph/crown-jewels/create-selector.png:
Similar comment as before, this text is going to be almost impossible to read when it's embedded in a docs page. Do we need the entire screenshot?

zmb3 commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,79 @@
+description: Describes how to use Teleport Policy Crown Jewels to see permission changes in Teleport.
+---
+
+With Teleport Policy's Access Graph, you can see change to users and resources permissions.
```
Do we need this first sentence? Feels like the first sentence should mention crown jewels in some way, and this one doesnt.

zmb3 commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,79 @@
+before and after snapshots of the permissions. This can help you identify when users get access to sensitive
+resources or when permissions are changed unexpectedly.
+
+This guide shows you how to configure Crown Jewels, how to mark resource and how to see permission changes.
```
```suggestion
This guide shows you how to configure Crown Jewels, how to mark resources as Crown Jewels, and how to see permission changes for these resources.
```

zmb3 commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,79 @@
+Access Graph is a feature of the [Teleport Policy](https://goteleport.com/platform/policy/) product
+available to Teleport Enterprise edition customers.
+
+After logging in to the Teleport UI, navigate to the Management tab. If enabled, Access Graph options can be found
+under the Permission Management section.
```
Note that this will need to be changed soon because Teleport 17 won't have a management tab.

If we can, it would be best to keep this description as generic as possible so it doesn't need to be rewritten if the UI changes.

r0mant commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,79 @@
+---
+
+With Teleport Policy's Access Graph, you can see change to users and resources permissions.
+Crown Jewels is a feature that allows you to mark resources and users as critical and see changes to them.
```
```suggestion
Crown Jewels allow you to mark resources and users as critical and see changes to them.
```

r0mant commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,79 @@
+
+With Teleport Policy's Access Graph, you can see change to users and resources permissions.
+Crown Jewels is a feature that allows you to mark resources and users as critical and see changes to them.
+Each time access for those resources or users changes, Teleport Policy will log the change and generate
```
What does "will log the change" mean? Is this referring to audit log or something else? I would be specific here.

r0mant commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,79 @@
+After logging in to the Teleport UI, navigate to the Management tab. If enabled, Access Graph options can be found
+under the Permission Management section.
+
+## Creating a Crown Jewel
```
Do we support defining crown jewels via IaC and/or CLI as well? If so, I would add this to the guide.

r0mant commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,79 @@
+Every time a change is made to a Crown Jewel, a new entry will be added to the list and an audit event
+will be generated in the Teleport Audit Log.
+
+### Required RBAC permissions
```
Feels like this should not be nested under "viewing permission changes"?
```suggestion
## Required RBAC permissions
```

r0mant commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,79 @@
+    - resources:
+      - crown_jewel
+      verbs:
+      - list
+      - read
+      - create
+      - delete
+      - update
```
Do we support wildcards here? I can't remember.
```suggestion
      - *
```

r0mant commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,79 @@
+![Create Matcher Name](../../../../img/access-graph/crown-jewels/create-matcher-name.png)
+
+The Crown Jewel will now be created, and you will see it in the list of Crown
+Jewels. Access Graph will now log changes to matched resource/user.
```
Similar to above, I'd change "will now log" to something more specific.

jakule replied:
Changed

r0mant commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,79 @@
+![Changes](../../../../img/access-graph/crown-jewels/changes.png)
+
+The generated changes have a diff format showing removed nodes with "-" and added nodes with "+".
+Every time a change is made to a Crown Jewel, a new entry will be added to the list and an audit event
```
Can we add details about the audit event? And add something regarding what they need to do if they want to get notified about these changes (e.g. using event handler until we build proper notification mechanism).

jakule replied:
Added

jakule committed:
Apply suggestions from code review

Co-authored-by: Roman Tkachenko <roman@goteleport.com>
Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

jakule committed:
Address code review issues

jakule committed:
Convert images from PNG to webp

jakule commented on docs/img/access-graph/crown-jewels/changes.png:
The total size of all assets in the initial PR was 1672 KB. I was able to cut half of that size with a PNG optimizer and get it down to 302 KB (~80%) by using webp.

In terms of the image itself, I wanted to show people how the page looks like. If your page is different, then you're probably on a wrong page. There is nothing specific that I want to highlight. I can remove the image if you think it doesn't add any value.

jakule committed:
Add audit event exporter section

zmb3 commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,99 @@
+description: Describes how to use Teleport Policy Crown Jewels to see permission changes in Teleport.
+---
+
+With Access Graph's Crown Jewel, you can see changes in user and resource permissions.
```
This still feels off to me. The title of the doc refers to _Teleport Policy Crown Jewels_ (it doesn't mention access graph at all).

How about this for an intro paragraph?

>>> Teleport Policy's Crown Jewel feature allows you to track changes to access for your most sensitive users or resources. When you mark a resource as a Crown Jewel, Teleport emits audit events any time access to that resource changes. These audit events include snapshots of the permissions before and after the change, which can alert you of unexpected access changes and allow you to verify the results.

jakule replied:
> The title of the doc refers to Teleport Policy Crown Jewels (it doesn't mention access graph at all).

We removed "access graph" from most places in our docs already. TBH I'm not sure why we did that, but I also try to refer to all features as "Teleport Policy" not "Access Graph".
I'll try to figure out why we did that. If we want to change the naming, we should probably also change the name in the Teleport UI to keep it consistent.

jakule replied:
>How about this for an intro paragraph?

LGTM. I updated the docs.

zmb3 commented on docs/pages/admin-guides/teleport-policy/integrations/crown-jewels.mdx:
```diff
@@ -0,0 +1,99 @@
+You can export the audit event using the event handler.
+The setup is described [here](../../management/export-audit-events.mdx).
+
+## Required RBAC permissions
```
This seems more like a prerequisite, doesn't it? If you don't have the required permissions you won't be able to do any of the actions described in this guide.

jakule replied:
In this case we should include the required RBAC role for every feature which we're currently not doing in any guide. I added it here, so people have an example role that can be added to an existing users. New TAG version also displays which RBAC permissions are required to use a feature when a user doesn't have them.

zmb3 replied:
Can we at least move it earlier in the guide?

It's a bit awkward to explain all the details about Crown Jewels, and then at the very end after the reader has made their way through the whole page we say "by the way make sure you have these permissions"

jakule replied:
Sure, done.

jakule committed:
Update Crown Jewels docs.

jakule committed:
Move required RBAC roles higher in the Crown Jewels docs

jakule commented:
Friendly ping @r0mant @ptgott

ptgott commented:
Hi @jakule, since there are already two review cycles for this PR, I am deferring to the current reviewers. Feel free to reach out if you have any questions. Thanks!

benarent reviewed (approved):
This PR looks good, my one suggestion would be to move it from /docs/admin-guides/teleport-policy/integrations/crown-jewels/ to  /docs/admin-guides/teleport-policy/crown-jewels/ since crown jewels is a core feature and not an integration.

jakule commented:
Friendly ping @zmb3 @r0mant

jakule commented:
> This PR looks good, my one suggestion would be to move it from /docs/admin-guides/teleport-policy/integrations/crown-jewels/ to /docs/admin-guides/teleport-policy/crown-jewels/ since crown jewels is a core feature and not an integration.

I was also thinking about it and I was 50/50 on it. Technically SSH key scanning is also a "core feature" and we added it as integration. In TAG we also added the SSH key scanning to the integration page to improve discoverability. I can move it if people think is makes more sense.

jakule committed:
Move crown jewels docs to Teleport Policy section.

jakule committed:
Rename Teleport Policy to Access Graph Crown Jewels

Updated references to the Crown Jewel feature from Teleport Policy to Access Graph in the title, description, and content.
